config/zephyr/Kconfig

```diff
@@ -22,14 +22,14 @@ menuconfig CHIP
 	imply REQUIRES_FULL_LIBC
 	imply NEWLIB_LIBC_NANO
 	imply CBPRINTF_LIBC_SUBSTS
-	imply POSIX_API if !ARCH_POSIX
```

Comment:
I'm not sure if Telink also wants to switch from `POSIX_API` to `NET_SOCKETS_POSIX_NAMES` so maybe it's better to make it conditional on our platform?
Unless @s07641069 confirms that Telink does want to follow this direction :)

Reply:
You are right, I added new kconfig to specify that nRF platform is built and implied it conditionally.